src/main/java/io/stargate/sgv2/jsonapi/service/resolver/AlterTableCommandResolver.java

```diff
@@ -0,0 +1,285 @@
+                          vectorizeConfig.modelName(),
+                          vectorizeConfig.authentication(),
+                          vectorizeConfig.parameters());
+                    }));
+    final AlterTableAttempt addColumnsAttempt =
+        new AlterTableAttemptBuilder(schemaObject).addColumns(addColumns).build();
+    if (!vectorizeConfigMap.isEmpty()) {
+      // Reading exising vectorize config from the table metadata
+      Map<String, String> existingExtensions = TableMetadataUtils.getExtensions(tableMetadata);
+      Map<String, VectorConfig.ColumnVectorDefinition.VectorizeConfig> existingVectorizeConfigMap =
+          TableMetadataUtils.getVectorizeMap(existingExtensions, objectMapper);
+
+      // Merge the new config to the existing vectorize config
+      existingVectorizeConfigMap.putAll(vectorizeConfigMap);
+
+      // New custom property to be updated
+      final Map<String, String> customProperties =
+          TableMetadataUtils.createCustomProperties(existingVectorizeConfigMap, objectMapper);
+      final AlterTableAttempt addVectorizeProperties =
+          new AlterTableAttemptBuilder(schemaObject).customProperties(customProperties).build();
+      // First execute the extension update for add columns
+      alterTableAttempts.add(addVectorizeProperties);
+    }
+    alterTableAttempts.add(addColumnsAttempt);
+
+    // Create the AlterData object
+    return alterTableAttempts;
+  }
+
+  private List<AlterTableAttempt> handleDropColumns(
+      AlterTableOperationImpl.DropColumns dc, TableSchemaObject schemaObject) {
+    List<AlterTableAttempt> alterTableAttempts = new ArrayList<>();
+    TableMetadata tableMetadata = schemaObject.tableMetadata();
+    List<String> dropColumns = dc.columns();
+    // Validate the columns to be dropped are present
+    List<CqlIdentifier> primaryKeys =
+        tableMetadata.getPrimaryKey().stream().map(ColumnMetadata::getName).toList();
+    for (String columnName : dropColumns) {
+      CqlIdentifier column = CqlIdentifierUtil.cqlIdentifierFromUserInput(columnName);
+      if (primaryKeys.contains(column)) {
+        throw SchemaException.Code.COLUMN_CANNOT_BE_DROPPED.get(
+            Map.of(
+                "reason",
+                "Primary key column `%s` cannot be dropped".formatted(column.asCql(true))));
+      }
+
+      if (tableMetadata.getColumn(column).isEmpty()) {
+        throw SchemaException.Code.COLUMN_NOT_FOUND.get(Map.of("column", column.asCql(true)));
+      }
+
+      final Optional<IndexMetadata> first =
+          tableMetadata.getIndexes().values().stream()
+              .filter(indexMetadata -> indexMetadata.getTarget().equals(column.asCql(true)))
+              .findFirst();
+      if (first.isPresent()) {
+        throw SchemaException.Code.COLUMN_CANNOT_BE_DROPPED.get(
+            Map.of(
+                "reason",
+                "Index exists on the column `%s`, drop `%s` index to drop the column"
```

Comment:
Should there not be property "column" to bind via template (and probably "index" as well)?

Reply:
Didn't get this, Have the column and index name binded as part of the reason text.

Reply:
Just asking why pre-format these in code, instead of using templating mechanism ErrorsV2 offers, like other error messages do.
Using templating mechanism could also allow us to include more metadata about failure itself  later on (if we included key/value pairs of information in JSON -- not something we do yet).

Reply:
One note: if it is not possible to use same templated message for all instance of Error, then fine to pre-format message to include via templating. But in general I think all variable parts that can be handled by yaml template should handled by it, and not by `String.format()`. This to keep templating in one place (ideally).